velox/functions/lib/Re2Functions.cpp

```diff
@@ -439,4 +518,74 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures() {
   };
 }
 
+std::shared_ptr<exec::VectorFunction> makeLike(
+    const std::string& name,
+    const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  auto numArgs = inputArgs.size();
+  VELOX_USER_CHECK(
+      numArgs == 2 || numArgs == 3,
+      "{} requires 2 or 3 arguments, but got {}",
+      name,
+      numArgs);
+
+  VELOX_USER_CHECK(
+      inputArgs[0].type->isVarchar(),
+      "{} requires first argument of type VARCHAR, but got {}",
+      name,
+      inputArgs[0].type->toString());
+
+  VELOX_USER_CHECK(
+      inputArgs[1].type->isVarchar(),
+      "{} requires second argument of type VARCHAR, but got {}",
+      name,
+      inputArgs[1].type->toString());
+
+  char escapeChar;
+  if (numArgs == 3) {
+    // TODO(aditi) : Validate that escapeChar is a single char.
+    VELOX_USER_CHECK(
+        inputArgs[2].type->isVarchar(),
+        "{} requires third argument of type VARCHAR, but got {}",
+        name,
+        inputArgs[2].type->toString());
+
+    BaseVector* constantEscape = inputArgs[2].constantValue.get();
+    VELOX_USER_CHECK(
+        constantEscape != nullptr && !constantEscape->isNullAt(0),
+        "{} requires third argument to be a constant of type VARCHAR",
+        name,
+        inputArgs[2].type->toString());
+
+    escapeChar = constantEscape->as<ConstantVector<char>>()->valueAt(0);
+  }
+
+  BaseVector* constantPattern = inputArgs[1].constantValue.get();
+  // TODO(aditi) : Process null patterns and escape char to return null
+  VELOX_USER_CHECK(
+      constantPattern != nullptr && !constantPattern->isNullAt(0),
+      "{} requires second argument to be a constant of type VARCHAR",
+      name,
+      inputArgs[1].type->toString());
+  auto pattern = constantPattern->as<ConstantVector<StringView>>()->valueAt(0);
+  return std::make_shared<LikeConstantPattern>(pattern, escapeChar);
+}
+
+std::vector<std::shared_ptr<exec::FunctionSignature>> likeSignatures() {
+  // varchar, varchar -> varchar
```

Comment:
should be varchar, varchar -> bool no?

Reply:
I misread the PrestoDB docs to interpret "like" as a vector function that returned the original column values. But a boolean operator does seem more preferable. Have changed this.